test/e2e/framework/cluster.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (C) 2024 ScyllaDB
+
+package framework
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cacheddiscovery "k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+)
+
+type ClusterInterface interface {
+	AdminClientInterface
+	DefaultNamespaceIfAny() (*corev1.Namespace, Client, bool)
+	CreateUserNamespace(ctx context.Context) (*corev1.Namespace, Client)
+}
+
+type createNamespaceFunc func(ctx context.Context, adminClient kubernetes.Interface, adminClientConfig *restclient.Config) (*corev1.Namespace, Client)
+
+type Cluster struct {
+	AdminClient
+
+	name string
+
+	createNamespace    createNamespaceFunc
+	defaultNamespace   *corev1.Namespace
+	defaultClient      Client
+	namespacesToDelete []*corev1.Namespace
+}
+
+var _ AdminClientInterface = &Cluster{}
+var _ ClusterInterface = &Cluster{}
+
+func NewCluster(name string, restConfig *restclient.Config, createNamespace createNamespaceFunc) *Cluster {
+	adminClientConfig := restclient.CopyConfig(restConfig)
+
+	return &Cluster{
+		AdminClient: AdminClient{
+			Config: adminClientConfig,
+		},
+
+		name: name,
+
+		createNamespace:  createNamespace,
+		defaultNamespace: nil,
+		defaultClient: Client{
+			Config: nil,
+		},
+		namespacesToDelete: []*corev1.Namespace{},
+	}
+}
+
+func (c *Cluster) DefaultNamespaceIfAny() (*corev1.Namespace, Client, bool) {
```

Comment:
I though you said you want it only on the framework - is this still pending changes or did I misinterpret your answer?

Reply:
`Namespace` func is only on the framework. `DefaultNamespaceIfAny` is there so that clusters can have a somewhat similar workflow, i.e. so that you can get namespace and its client of the default cluster from the test (for other clusters you'd just do `CreateUserNamespace`). Otherwise the only way to get it would be through framework, so it wouldn't align with the other clusters' workflow. It's a tradeoff at this point, but it sucks a bit one way or another.